<!DOCTYPE html>
<html lang="en">
    <head>
        <title>PASA Overview</title>

            <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css">
            <!--<link rel="stylesheet" href="https://gitcdn.github.io/bootstrap-toggle/2.2.2/css/bootstrap-toggle.min.css">-->
            <link rel="stylesheet" href="css/general.css">
            <link rel="stylesheet" href="css/nav.css">
            <link rel="shortcut icon" type="image/x-icon" href="pics/logo.png" />

            <!--files for the select tag-->
            <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/css/bootstrap-select.min.css">
            <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-select/1.12.4/js/bootstrap-select.min.js"></script>

            <!--files for the checkbox tag (toggle)-->
            <script src="https://gitcdn.github.io/bootstrap-toggle/2.2.2/js/bootstrap-toggle.min.js"></script>

            <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
            <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
    </head>
    <body>
        <nav role="navigation" class="navbar navbar-inverse navbar-fixed-top" id="nav">
            <div class="jumbotron" id="jumbo">
                <div class="container">
                    <div class="row" id="title-row">
                        <div class="col-md-1">
                        </div>
                        <div class="col-md-1">
                            <img src="pics/logo.png" id="antibody_image" class="img-rounded">
                        </div>
                        <div class="col-md-9">
                            <span id="pasa-title">PASA</span>&nbsp;&nbsp;&nbsp;&nbsp;<span id="sub-title"><b>Proteomics Analysis of Serum Antibodies </b></span><br>
                        </div>
                    </div>
                </div>
            </div>
            <div class="container" id="nav_container">
                <div class="row">
                    <div class="col-md-1">
                    </div>
                    <div class="col-md-8">
                        <ul class="nav navbar-nav" id="menu-nav">
                            <li>
                                <a href="index.html">Home</a>
                            </li>
                            <li class="active">
                                <a href="overview.html">Overview</a>
                            </li>
                            <li>
                                <a href="gallery.html">Gallery</a>
                            </li>
                            <li>
                                <a href="terms.html">Terms of Use</a>
                            </li>
                            <li>
                                <a href="credits.html">Credits</a>
                            </li>
                            <li>
                                <a href="https://english.tau.ac.il/" id="tau-link" target="_blank">
                                    <img src=pics/TAU_LOGO.png id="tau-logo">
                                </a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </nav>
        <div id="behind-nav-bar">
        </div>
            <div class="container" id="overview_container">
            <div class="row">
                <div class="col-md-1">
                </div>
                <div class="col-md-10" align="justify">
                    <h2>Introduction</h2>
                    Shotgun high-resolution proteomics antibodies is necessary for understanding the
                    dynamics and composition of polyclonal antibodies circulating in
                    serum [<a href="https://pubs.acs.org/doi/abs/10.1021/ac4037679">1</a>]. At steady
                    state circulating antibodies are produced by terminally differentiated B cells but after
                    exposure to antigen the humoral response generates a diverse antibody repertoire
                    due to SHM and V(D)J recombination [<a href="https://www.pnas.org/content/110/8/2993">2</a>,
                    <a href="https://www.ncbi.nlm.nih.gov/pubmed/24469811">3</a>] . For that reason, it is important to unveil
                    whether the Igs encoded by the peripheral blood B cells correspond to antibodies
                    present in circulation. Moreover, addressing questions as what are the relative
                    concentrations of serum antigen-specific monoclonal antibodies (within the
                    polyclonal pool), what are their repertoire features (e.g. V(D)J family usage, CDR3
                    length etc.) is important in order to shed light on mechanisms related to the
                    development of the immune response.

                    <br><br>

                    <h2>What is PASA?</h2>
                    PASA web server is user-friendly, free and open to all users and there is no login
                    requirement. PASA allows generating quantitative molecular measurements related
                    to serum antibody repertoire from high-resolution proteomic data. The PASA
                    web server is applicable for researchers at all levels that are interested in addressing
                    basic questions related to B cell development and differentiation, as well as an
                    application focused tool for vaccine development, immunodiagnostic discovery and
                    monoclonal antibody engineering.

                    <br><br>

                    <h2>Input</h2>
                    The web server accepts proteomics spectral files in raw format, up to 3 technical
                    replicates to each fraction elution/flow through. For the integration with data derived
                    from NGS of antibody encoding B cells the user is requested to upload V gene
                    sequence database at the amino acid level in Fasta format. This database can be
                    generated from our recently published ASAP web server [<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6077260">4</a>] that facilitates the analysis of
                    sequences derived from NGS (see <a href="http://asap.tau.ac.il">http://asap.tau.ac.il</a> for more details).

                    <br><br>
                    <img src="pics/flow.jpg" id="graphical_abstract">
                    <br><br>

                    <h2>Output and methodology<br></h2>
                    By far the highest diversity in antibodies occurs within the CDRH3 region
                    [<a href="https://www.ncbi.nlm.nih.gov/pubmed/10933393">5</a>] , which is
                    overwhelmingly responsible for antigen recognition. The CDRH3 is thus considered
                    as a unique identifier of antibody clones and hence is the focus of our analysis. Our
                    methodology is divided into two parts. The first part utilizes the MaxQuant
                    [<a href="https://www.ncbi.nlm.nih.gov/pubmed/19029910">6</a> ,
                    <a href="https://www.ncbi.nlm.nih.gov/pubmed/24942700">7</a>] a proteomic search algorithm that is used by
                    PASA to identify the sequences of peptides derived from serum
                    antibodies. PASA accepts up to three technical replicates of
                    proteomic spectral files from LC-MS/MS of antibodies that were
                    enriched by affinity chromatography against a given antigen
                    (elution) and up to three replicates of proteomic spectral files from
                    LC-MS/MS of antibodies that were depleted from the antigen
                    specific antibodies (flow through). The output from the first step is a
                    list of high confidence peptide sequences that are mapped to one of
                    the antibody sequences in the NGS database.
                    <br><br>

                    The output of this step is summarized in a file named:<br><br>
                    1. list_of_curated_peptides.txt (accessed via the link in the results
                    section with the same name of the file name) and lists the identified
                    peptide sequences that map to one of the antibody V genes in the
                    uploaded database. The header of the columns in this file includes
                    the following:</li>
                    <br><br>

                    <li>Peptide_name - the identified peptide sequence that was mapped to an antibody sequence in the uploaded database. Only peptides that received high score by MaxQuant are included in this list.</li>
                    <li>Average_Intensity_elution - the average intensity of peptides from all replicates in the elution. Peptides identified in the elution fraction are marked as peptides derived from antibodies that are antigen-specific.</li>
                    <li>Average_enrichment_elution - the average enrichment in the elution represents the relative abundance of the peptide in all elution replicates.</li>
                    <li>Average_Intensity_flowthrough - the average intensity of peptides from all replicates in the flowthrough fruction. Peptides identified in the flowthrough fraction are marked as peptides derived from antibodies that are not antigen-specific.</li>
                    <li>Average_enrichment_flowthrough - the average enrichment in the elution represents the relative abundance of the peptide in all flowthrough replicates.</li>
                    <br>
                    In the second part, the identified peptides are filtered to be mapped to antibodies and are stratified as follows: <br>
                    2. informative_CDR3_peptides - peptides that map uniquely to a specific antibody colonotype and are derived from the CDR3 region of the antibody. These peptides are used to provide the final list of identified antibodies.<br>
                    3. informative_non_CDR3_peptides - peptides that map uniquely to a specific clonotype but to a region in the antibody variable region but not to CDR3 region.<br>
                    4. Non_informative_peptides - peptides that are promiscuous thus, do not map to a unique antibody clonotype and cannot be used to identify a specific antibody.<br>
                    <br>
                    Each group of peptides are listed in a Tab Separated Values (.tsv) file and can be accessed through the links with matching names (i.e. list of informative CDR3 peptides, list of informative NON CDR3 peptides and list of NON informative peptides).
                    <br><br>
                    Both informative peptide files include the columns as follows:
                    <li>Peptide_Seqeunce - the sequence of the identified peptide</li>
                    <li>Db_sequence - the list of full antibody variable region sequence that were mapped by the peptide</li>
                    <li>Sequence_header - the header of the variable region sequence as appears in the uploaded database</li>
                    <li>CDR3 - the CDR3 sequence of the identified antibody clonotype</li>
                    <li>V_family - the V family of the identified variable region of the antibody</li>
                    <li>D_family - the D family of the identified variable region of the antibody. In case the identification is of a variable region of a light chain, this field will hold the NA value.</li>
                    <li>J_family - the J family of the identified variable region of the antibody</li>
                    <li>Chain_type - the antibody variable region chain type (i.e. IGH, IGL, IGK)</li>
                    NON informative peptide file includes a list of identified peptides without additional information as they do not contribute to the identification of non-promiscuous antibodies.<br>
                    <br>
                    5. CDR3 IGH length distribution - present a histogram of the distribution of CDR3 length from antibody heavy chains as identified by proteomic informative CDR3 peptides (see example in gallery).
                    <br><br>
                    6. CDR3 IGL length distribution - present a histogram of the distribution of CDR3 length from antibody light chains as identified by proteomic informative CDR3 peptides (see example in gallery).
                    <br><br>
                    7. The 10 links following the above result links include the distribution of V(D)J family usage and the combinations (e.g. VD, VJ) of the antibody variable regions that were identified by proteomic informative CDR3 peptides.
                    <br><br>
                    8. Isotype distribution - a pie chart showing the distribution of the antibodies that were identified by proteomic informative CDR3 peptides (see gallery for example).
                    <br><br>
                    9. Proteomics vs Genetics - this summary chart plots the relationship between the relative abundance of proteomically identified antibodies (serum antibodies) to their relative abundance as determined by NGS of the variable region of B cells.
                    <br>

                    <br>



                    <br><br>

                    <h2>References</h2>
                    <li><a href="https://pubs.acs.org/doi/abs/10.1021/ac4037679" target="_blank" id="ref1">Boutz, D. R. et al. Proteomic identification of monoclonal antibodies from serum. Anal. Chem. (2014). doi:10.1021/ac4037679.</a></li>
                    <br>
                    <li><a href="https://www.pnas.org/content/110/8/2993" target="_blank" id="ref2">Wine, Y. et al. Molecular deconvolution of the monoclonal antibodies that comprise the polyclonal serum response. Proc. Natl. Acad. Sci. (2013). doi:10.1073/pnas.1213737110.</a></li>
                    <br>
                    <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/24469811" target="_blank" id="ref3">Lavinder, J. J. et al. Identification and characterization of the constituent human serum antibodies elicited by vaccination. Proc. Natl. Acad. Sci. (2014). doi:10.1073/pnas.1317793111.</a></li>
                    <br>
                    <li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6077260" target="_blank" id="ref4">Avram, O. et al. ASAP - A web server for Immunoglobulin-Sequencing Analysis Pipeline. Front. Immunol. 9, 1â€“12 (2018)..</a></li>
                    <br>
                    <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/10933393" target="_blank" id="ref5">Xu, J. L. & Davis, M. M. Diversity in the CDR3 region of V(H) is sufficient for most antibody specificities. Immunity (2000). doi:10.1016/S1074-7613(00)00006-6.</a></li>
                    <br>
                    <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/19029910" target="_blank" id="ref6">Cox, J. & Mann, M. MaxQuant enables high peptide identification rates, individualized p.p.b.-range mass accuracies and proteome-wide protein quantification. Nat. Biotechnol. (2008). doi:10.1038/nbt.1511.</a></li>
                    <br>
                    <li><a href="https://www.ncbi.nlm.nih.gov/pubmed/24942700" target="_blank" id="ref7">Cox, J. et al. Accurate Proteome-wide Label-free Quantification by Delayed Normalization and Maximal Peptide Ratio.</a></li>


                    <br><br><br><br>
                </div>
            </div>
        </div>
        <nav role="navigation" class="navbar-inverse navbar-fixed-bottom ng-binding" style="color: #cccccc; text-align: center;">
            <h4>
                    Research Site | <a href="https://www.tau.ac.il/~talp/" target="_blank">Pupko Group</a>
            </h4>
        </nav>
    </body>
</html>







